core/src/main/java/com/valtech/aem/saas/core/util/page/ContainingPage.java

```diff
@@ -0,0 +1,27 @@
+package com.valtech.aem.saas.core.util.page;
+
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+import com.valtech.aem.saas.core.util.LoggedOptional;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+
+@Slf4j
+@RequiredArgsConstructor
+public final class ContainingPage {
```

Comment:
- [ ] please remove, no util classes for complex objects.

Reply:
This is a reusable logic that most probably will be needed in other implementations as well. I'd say we leave it be and move it to another package called 'common' (meaning the logic in this package could certainly be reused in future implementations)